test/unit/plugins/guests/redhat/cap/change_host_name_test.rb

```diff
@@ -31,13 +31,40 @@
       expect(comm.received_commands[1]).to match(/\/etc\/sysconfig\/network-scripts\/ifcfg/)
       expect(comm.received_commands[1]).to match(/hostnamectl set-hostname --static '#{name}'/)
       expect(comm.received_commands[1]).to match(/hostnamectl set-hostname --transient '#{name}'/)
-      expect(comm.received_commands[1]).to match(/service network restart/)
+      expect(comm.received_commands[1]).to match(/service network restart|systemctl restart NetworkManager.service/)
     end
 
     it "does not change the hostname if already set" do
       comm.stub_command("hostname -f | grep '^#{name}$'", exit_code: 0)
       cap.change_host_name(machine, name)
       expect(comm.received_commands.size).to eq(1)
     end
+
+    context "restarts the network" do
```

Comment:
Hey @briancain - @jdoss asked if I minded jumping into this branch to assist with some tests. I'm a coworker of his who works in testing. I was hoping to get your input on a flaw that I know exists with my tests here.  As you can see below, I'm stubbing the commands to comm that @jdoss introduced and I'm returning 0 or 1 exit_codes depending on the logic I want to test.  The problem I have is that all three assertions will pass, regardless of which of the three commands get stubbed and their return codes.  The reason for this is that `comm.received_commands[1]` is always the entire bash string that gets built up in the .change_host_name method.  I'm not sure I see at the moment how I can assert that comm behaved differently when it received `"test -f /etc/init.d/network"` vs the other commands.  Is there a way that I can verify that comm sent off the `systemctl restart NetworkManager.service` message for example?  I searched in the other plugin test files for precedent, but I'm not seeing anywhere where we test embedded bash IF logic in this way.  I could totally be wrong!  Wondering if you have any insight?

Reply:
Awesome - happy to help!